File: FileViewer.Model/StoredFile.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace FileViewer.Model
{
    public record StoredFile
    {
        [BsonId]
        public ObjectId id { get; set; }

        [BsonElement("name")]
        public string name { get; set; } = "";

        [BsonElement("type")]
        public string type { get; set; } = "";

        [BsonElement("size")]
        public int size { get; set; } = 0;

        [BsonElement("data")]
        public string data { get; set; } = "";

        [BsonElement("createdAt")]
        public DateTime createdAt { get; set; }

        [BsonElement("isDeleted")]
        public bool isDeleted { get; set; }
    }
}

File: fileviewer-web-client/fileviewer-web-client/src/model/stored-file.ts
export class StoredFile {
    name = "";
    size = 0;
    type = "";
    data = "";
    createdAt = Date.now;
    isDeleted = false;
    isViewable = true;
    constructor (file : File){
        this.name = file.name;
        this.size = file.size;
        this.type = file.type;
    }
}
